tf2_ros/src/buffer_server.cpp

```diff
@@ -143,6 +143,34 @@ rclcpp_action::CancelResponse BufferServer::cancelCB(GoalHandle gh)
   return rclcpp_action::CancelResponse::REJECT;
 }
 
+void BufferServer::serviceCB(const std::shared_ptr<LookupTransformService::Request> request,
+          std::shared_ptr<LookupTransformService::Response> response)
+{
+    // TODO: implement retrying + timeout
+    try {
+      response->transform = lookupTransform(request);
+    } catch (const tf2::ConnectivityException & ex) {
+      response->error.error = response->error.CONNECTIVITY_ERROR;
+      response->error.error_string = ex.what();
+    } catch (const tf2::LookupException & ex) {
+      response->error.error = response->error.LOOKUP_ERROR;
+      response->error.error_string = ex.what();
+    } catch (const tf2::ExtrapolationException & ex) {
+      response->error.error = response->error.EXTRAPOLATION_ERROR;
+      response->error.error_string = ex.what();
+    } catch (const tf2::InvalidArgumentException & ex) {
+      response->error.error = response->error.INVALID_ARGUMENT_ERROR;
+      response->error.error_string = ex.what();
+    } catch (const tf2::TimeoutException & ex) {
+      response->error.error = response->error.TIMEOUT_ERROR;
+      response->error.error_string = ex.what();
+    } catch (const tf2::TransformException & ex) {
+      response->error.error = response->error.TRANSFORM_ERROR;
+      response->error.error_string = ex.what();
+    }
+  
+}
```

Comment:
```suggestion
void BufferServer::serviceCB(
  const std::shared_ptr<LookupTransformService::Request> request,
  std::shared_ptr<LookupTransformService::Response> response)
{
  // TODO: implement retrying + timeout
  try {
    response->transform = lookupTransform(request);
  } catch (const tf2::ConnectivityException & ex) {
    response->error.error = response->error.CONNECTIVITY_ERROR;
    response->error.error_string = ex.what();
  } catch (const tf2::LookupException & ex) {
    response->error.error = response->error.LOOKUP_ERROR;
    response->error.error_string = ex.what();
  } catch (const tf2::ExtrapolationException & ex) {
    response->error.error = response->error.EXTRAPOLATION_ERROR;
    response->error.error_string = ex.what();
  } catch (const tf2::InvalidArgumentException & ex) {
    response->error.error = response->error.INVALID_ARGUMENT_ERROR;
    response->error.error_string = ex.what();
  } catch (const tf2::TimeoutException & ex) {
    response->error.error = response->error.TIMEOUT_ERROR;
    response->error.error_string = ex.what();
  } catch (const tf2::TransformException & ex) {
    response->error.error = response->error.TRANSFORM_ERROR;
    response->error.error_string = ex.what();
  } 
}
```

Reply:
Also, what's the deal with the TODO comment about retrying and timeout?  Do we need to call a different `lookupTransform` signature?

Reply:
retrying - I must say I don't remember what I meant with that....

for the timeout: there is the `timeout` field in the `LookupTransform.srv` (as well as in the action). I was wondering if I could just pass it to `buffer_.lookupTransform` but I don't really understand what happens if the timeout is reached - I don't see an exception being raised to catch it and fill it in `response->error.error`. 
For the action, the timeout behavior is implemented manually, see `end_time`.